### ISSUE TITLE
[Core][Pet] Owner combat state does not influence pet combat state

### ISSUE THREAD

**Issue by nevrozac** [Stormforge-gg/Bugtracker-Fordring#1707]
**Description**: 
When a player enters combat his pet should also enter combat with the same target(s). As seen in retail video here (start at 3:07) pet will enter combat when owner does and no longer be able to cast spells that can be cast only out of combat (lesser invisibility in this case)
https://youtu.be/JNsWnoCH5Ww?t=187
**Current behaviour**:
pets have independent combat state and don't even generally attack players targets when hes doing damage to them
**Expected behaviour**: 
pets should be forcibly linked with players in regards to combat state

**Comment by Kripp0** [comment]
Retail video from 2014?

**Comment by nevrozac** [comment]
You're right this video from that channel is no longer retail. Regardless - evidence of video should not be necessary on this issue

**Comment by haephaistoss** [comment]
it's old AT, but the statement is true.

Warlocks could put themselves in combat by casting hellfire and soul linked hellfire dmg to the pet would cause combat for 5 seconds to both of them with no actual enemy being hit

**Comment by Kripp0** [comment]
https://youtu.be/RtkraBQbpgI?t=127 - no combat > send pet to attack > still no combat = patch 3.2.2.

**Comment by haephaistoss** [comment]
https://www.reddit.com/r/worldofpvp/comments/95yz3x/entering_combat_from_pet_attack/

Prolly bugged, no change is visible regarding these anywhere, so any goofy mechanic you find is a result of that.

**Comment by nevrozac** [comment]
The issue I'm posting about is that pet combat state is not directly influenced by owner combat state, not the other way around, even tho that should get looked at as well.

I'm posting because pets like succubus should not attempt to go invisible or others in stealth when their owner is in combat. (issues which are very gamebreaking in certain aspects of the game)
Please keep the bug report on topic

**Comment by nevrozac** [comment]
Another type of issue caused by this bug is that since pets combat state is not linked to owners, then a succubus can start casting invisibility and become untargettable after a while of not being damaged even though its warlock has never exited combat, giving the warlock a very unfair advantage
